lldb/source/Plugins/LanguageRuntime/Swift/SwiftLanguageRuntimeDynamicTypeResolution.cpp

```diff
@@ -1819,7 +1819,16 @@ bool SwiftLanguageRuntimeImpl::GetDynamicTypeAndAddress_Protocol(
   auto &target = m_process.GetTarget();
   assert(IsScratchContextLocked(target) &&
          "Swift scratch context not locked ahead");
-  auto &remote_ast = GetRemoteASTContext(scratch_ctx);
+
+  auto *tss =
+      llvm::dyn_cast_or_null<TypeSystemSwift>(protocol_type.GetTypeSystem());
+  if (!tss)
+    return false;
+
+  const swift::reflection::TypeRef *protocol_typeref =
+      GetTypeRef(protocol_type, tss->GetSwiftASTContext());
+  if (!protocol_typeref)
+    return false;
```

Comment:
on first glance, there are two new ways this function could fail. Should we log? Are there known cases where these can happen, or is this needed as a fall out of these APIs returning pointers?

Reply:
Good idea, I'll add some logging to these failure points!